<ui version="4.0" >
 <class>Pkcs11Config</class>
 <widget class="QDialog" name="Pkcs11Config" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>682</width>
    <height>555</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>PKCS#11 Module Configuration</string>
  </property>
  <layout class="QVBoxLayout" >
   <property name="margin" >
    <number>9</number>
   </property>
   <property name="spacing" >
    <number>6</number>
   </property>
   <item>
    <widget class="QGroupBox" name="gb_general" >
     <property name="title" >
      <string>General</string>
     </property>
     <layout class="QHBoxLayout" >
      <property name="margin" >
       <number>9</number>
      </property>
      <property name="spacing" >
       <number>6</number>
      </property>
      <item>
       <layout class="QVBoxLayout" >
        <property name="margin" >
         <number>0</number>
        </property>
        <property name="spacing" >
         <number>6</number>
        </property>
        <item>
         <widget class="QCheckBox" name="ck_allowroot" >
          <property name="text" >
           <string>Allow Loading of Root CAs</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="ck_allowprotected" >
          <property name="text" >
           <string>Allow Protected Authentication</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" >
          <property name="margin" >
           <number>0</number>
          </property>
          <property name="spacing" >
           <number>6</number>
          </property>
          <item>
           <widget class="QLabel" name="label_3" >
            <property name="text" >
             <string>Log Level:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="sb_loglevel" />
          </item>
          <item>
           <spacer>
            <property name="orientation" >
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" >
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QGroupBox" name="gb_pincache" >
        <property name="title" >
         <string>Remember PIN</string>
        </property>
        <property name="checkable" >
         <bool>true</bool>
        </property>
        <property name="checked" >
         <bool>false</bool>
        </property>
        <layout class="QVBoxLayout" >
         <property name="margin" >
          <number>9</number>
         </property>
         <property name="spacing" >
          <number>6</number>
         </property>
         <item>
          <widget class="QRadioButton" name="rb_pincache_nolimit" >
           <property name="text" >
            <string>Until Session Ends</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" >
           <property name="margin" >
            <number>0</number>
           </property>
           <property name="spacing" >
            <number>6</number>
           </property>
           <item>
            <widget class="QRadioButton" name="rb_pincache_time" >
             <property name="text" >
              <string>For a Time (in Seconds):</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="sb_pincache_time" >
             <property name="maximum" >
              <number>86400</number>
             </property>
            </widget>
           </item>
           <item>
            <spacer>
             <property name="orientation" >
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" >
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="gb_modules" >
     <property name="title" >
      <string>PKCS#11 Modules</string>
     </property>
     <layout class="QHBoxLayout" >
      <property name="margin" >
       <number>9</number>
      </property>
      <property name="spacing" >
       <number>6</number>
      </property>
      <item>
       <layout class="QVBoxLayout" >
        <property name="margin" >
         <number>0</number>
        </property>
        <property name="spacing" >
         <number>6</number>
        </property>
        <item>
         <widget class="QListView" name="lv_modules" />
        </item>
        <item>
         <layout class="QHBoxLayout" >
          <property name="margin" >
           <number>0</number>
          </property>
          <property name="spacing" >
           <number>6</number>
          </property>
          <item>
           <widget class="QPushButton" name="pb_add" >
            <property name="text" >
             <string>Add</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer>
            <property name="orientation" >
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" >
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="pb_remove" >
            <property name="text" >
             <string>Remove</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" >
        <property name="margin" >
         <number>0</number>
        </property>
        <property name="spacing" >
         <number>6</number>
        </property>
        <item>
         <widget class="QGroupBox" name="gb_details" >
          <property name="title" >
           <string>Details</string>
          </property>
          <layout class="QVBoxLayout" >
           <property name="margin" >
            <number>9</number>
           </property>
           <property name="spacing" >
            <number>6</number>
           </property>
           <item>
            <layout class="QHBoxLayout" >
             <property name="margin" >
              <number>0</number>
             </property>
             <property name="spacing" >
              <number>6</number>
             </property>
             <item>
              <widget class="QLabel" name="label_2" >
               <property name="text" >
                <string>Name:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="le_name" >
               <property name="minimumSize" >
                <size>
                 <width>200</width>
                 <height>0</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" >
             <property name="margin" >
              <number>0</number>
             </property>
             <property name="spacing" >
              <number>6</number>
             </property>
             <item>
              <widget class="QLabel" name="lb_file" >
               <property name="text" >
                <string>Module File:</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" >
               <property name="margin" >
                <number>0</number>
               </property>
               <property name="spacing" >
                <number>6</number>
               </property>
               <item>
                <widget class="QLineEdit" name="le_library" >
                 <property name="minimumSize" >
                  <size>
                   <width>250</width>
                   <height>0</height>
                  </size>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="pb_browse" >
                 <property name="text" >
                  <string>Browse</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QCheckBox" name="ck_modallowprotected" >
             <property name="text" >
              <string>Allow Protected Authentication</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="ck_certprivate" >
             <property name="text" >
              <string>Certificates Stored As Private Objects</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="gb_keymodes" >
             <property name="title" >
              <string>Custom Private Key Modes</string>
             </property>
             <property name="checkable" >
              <bool>true</bool>
             </property>
             <property name="checked" >
              <bool>false</bool>
             </property>
             <layout class="QGridLayout" >
              <property name="margin" >
               <number>9</number>
              </property>
              <property name="spacing" >
               <number>6</number>
              </property>
              <item row="1" column="1" >
               <widget class="QCheckBox" name="ck_modeunwrap" >
                <property name="text" >
                 <string>Use Unwrap</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1" >
               <widget class="QCheckBox" name="ck_modedecrypt" >
                <property name="text" >
                 <string>Use Decrypt</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0" >
               <widget class="QCheckBox" name="ck_modesignrecover" >
                <property name="text" >
                 <string>Use Sign Recover</string>
                </property>
               </widget>
              </item>
              <item row="0" column="0" >
               <widget class="QCheckBox" name="ck_modesign" >
                <property name="text" >
                 <string>Use Sign</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" >
             <property name="margin" >
              <number>0</number>
             </property>
             <property name="spacing" >
              <number>6</number>
             </property>
             <item>
              <widget class="QLabel" name="label" >
               <property name="text" >
                <string>Slot Event Method:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="cb_slotmethod" >
               <item>
                <property name="text" >
                 <string>Auto</string>
                </property>
               </item>
               <item>
                <property name="text" >
                 <string>Trigger</string>
                </property>
               </item>
               <item>
                <property name="text" >
                 <string>Poll</string>
                </property>
               </item>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" >
             <property name="margin" >
              <number>0</number>
             </property>
             <property name="spacing" >
              <number>6</number>
             </property>
             <item>
              <widget class="QCheckBox" name="ck_poll" >
               <property name="text" >
                <string>Poll timeout:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="sb_poll" />
             </item>
             <item>
              <spacer>
               <property name="orientation" >
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" >
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer>
          <property name="orientation" >
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" >
           <size>
            <width>20</width>
            <height>21</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox" >
     <property name="orientation" >
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons" >
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::NoButton|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>ck_allowroot</tabstop>
  <tabstop>ck_allowprotected</tabstop>
  <tabstop>sb_loglevel</tabstop>
  <tabstop>gb_pincache</tabstop>
  <tabstop>rb_pincache_nolimit</tabstop>
  <tabstop>rb_pincache_time</tabstop>
  <tabstop>sb_pincache_time</tabstop>
  <tabstop>lv_modules</tabstop>
  <tabstop>pb_add</tabstop>
  <tabstop>pb_remove</tabstop>
  <tabstop>le_name</tabstop>
  <tabstop>le_library</tabstop>
  <tabstop>pb_browse</tabstop>
  <tabstop>ck_modallowprotected</tabstop>
  <tabstop>ck_certprivate</tabstop>
  <tabstop>gb_keymodes</tabstop>
  <tabstop>ck_modesign</tabstop>
  <tabstop>ck_modesignrecover</tabstop>
  <tabstop>ck_modedecrypt</tabstop>
  <tabstop>ck_modeunwrap</tabstop>
  <tabstop>cb_slotmethod</tabstop>
  <tabstop>ck_poll</tabstop>
  <tabstop>sb_poll</tabstop>
  <tabstop>buttonBox</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>Pkcs11Config</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>336</x>
     <y>185</y>
    </hint>
    <hint type="destinationlabel" >
     <x>320</x>
     <y>137</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>Pkcs11Config</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>361</x>
     <y>183</y>
    </hint>
    <hint type="destinationlabel" >
     <x>363</x>
     <y>135</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
